cpp/src/arrow/record_batch.h

```diff
@@ -310,7 +310,20 @@ class ARROW_EXPORT RecordBatchReader {
   /// \brief Read the next record batch in the stream. Return null for batch
   /// when reaching end of stream
   ///
-  /// \param[out] batch the next loaded batch, null at end of stream
+  /// example:
+  /// ```
+  /// std::shared_ptr<RecordBatch> batch;
+  /// while (true) {
+  ///   ARROW_RETURN_NOT_OK(reader->ReadNext(&batch));
+  ///   if (batch == nullptr) {
```

Comment:
Should we add a block to suggest skipping empty batch?

```
if (batch->num_rows() == 0) {
  continue;
}
```

Reply:
Personally I found some place handled this( like DatasetWriter ) but some not...